prime-router/src/main/kotlin/JurisdictionalFilters.kt

```diff
@@ -157,7 +161,112 @@ class OrEquals : JurisdictionalFilter {
     }
 }
 
+
+/**
+ * A filter that filter nothing -- allows all data through
+ */
+class AllowAll : JurisdictionalFilter {
+    override val name = "allowAll"
+
+    override fun getSelection(args: List<String>, table: Table, receiver: Receiver): Selection {
+        // On empty args (eg, "allowAll()"), our regex returns args of size 1, with a single empty string.
+        // Didn't bother trying to fix the regex.
+        if (args.size > 1) error("For rcvr ${receiver.fullName} Expecting no args for filter $name." +
+            " Got ${args.joinToString(",")}")
+        return Selection.withRange(0, table.rowCount())
+    }
+}
+
+
+/**
+ * Implements a quality check match.  If a row has valid data for all the columns, the row is selected.
+ * If any column name does not exist, nothing passes thru the filter.
+ * hasValidDataFor(columnName1, columnName2, columnName3, ...)
+ * If no columns are passed, all rows are selected.  So, any number of args is acceptable.
+ */
+class HasValidDataFor : JurisdictionalFilter {
+    override val name = "hasValidDataFor"
+
+    override fun getSelection(args: List<String>, table: Table, receiver: Receiver): Selection {
+        var selection = Selection.withRange(0, table.rowCount())
+
+        val columnNames = table.columnNames()
+        args.forEach {
+            if (columnNames.contains(it)) {
+                selection = selection.andNot(table.stringColumn(it).isEmptyString)
+            } else {
```

Comment:
For now we're just checking for the existence of data, but at some point we may want to extend this to do things like check for valid CLIA, valid zip, etc.

Reply:
I'm assuming that invalid data has gotten "zeroed out" - that is, at this point the field is empty.  But I didn't test this exact situation, I admit.